tests/test_connect.py

```diff
@@ -1736,8 +1738,104 @@ async def test_no_explicit_close_with_debug(self):
                                        r'unclosed connection') as rw:
                 await self._run_no_explicit_close_test()
 
-            msg = rw.warning.args[0]
+            msg = " ".join(rw.warning.args)
             self.assertIn(' created at:\n', msg)
             self.assertIn('in test_no_explicit_close_with_debug', msg)
         finally:
             self.loop.set_debug(olddebug)
+
+
+class TestConnectionAttributes(tb.HotStandbyTestCase):
+
+    async def _run_connection_test(
+        self, connect, target_attribute, expected_port
+    ):
+        conn = await connect(target_session_attribute=target_attribute)
+        self.assertTrue(_get_connected_host(conn).endswith(expected_port))
+        await conn.close()
+
+    async def test_target_server_attribute_port(self):
+        if self.master_cluster.get_pg_version()[0] == 11:
+            self.skipTest("PostgreSQL 11 seems to have issues with this test")
```

Comment:
Please elaborate.  If tests on 11 are failing then we either need a version-specific workaround or disable the feature on those servers.

Reply:
This is a failing test that only happens on version 11:
https://github.com/JesseDeLoore/asyncpg/actions/runs/4043448766/jobs/6952375723#step:7:342

Not sure what to make of it.

Reply:
To elaborate: It looks like "SELECT pg_catalog.pg_is_in_recovery()" returns False for the hot standby server in PG11.
Pretty sure this is not intended. Do you prefer disabling the feature for this version or trying to find a workaround (although I wouldn't know where to look for that).

Reply:
I'll see if I can repro locally.

Reply:
Were you able to reproduce it locally ? Anything else I can do to help move this along ?

Reply:
Hi. Gentle reminder. If there is anything I can do, do let me know.

Reply:
Hi, are there any updates? Is there anything I can do ?

Reply:
This diff fixes the test on PG 11:

```
diff --git a/asyncpg/cluster.py b/asyncpg/cluster.py
index 0999e41..4467cc2 100644
--- a/asyncpg/cluster.py
+++ b/asyncpg/cluster.py
@@ -626,7 +626,7 @@ class HotStandbyCluster(TempCluster):
                 'pg_basebackup init exited with status {:d}:\n{}'.format(
                     process.returncode, output.decode()))

-        if self._pg_version <= (11, 0):
+        if self._pg_version < (12, 0):
             with open(os.path.join(self._data_dir, 'recovery.conf'), 'w') as f:
                 f.write(textwrap.dedent("""\
                     standby_mode = 'on'
```

Please apply and rebase, and I think this is good to go.

Reply:
I applied and merged master. Is that ok for you, or do you require an actual rebase?